core/src/main/java/inetsoft/web/viewsheet/controller/annotation/VSAnnotationFormatService.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * This file is part of StyleBI.
+ * Copyright (C) 2025  InetSoft Technology
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package inetsoft.web.viewsheet.controller.annotation;
+
+import inetsoft.cluster.*;
+import inetsoft.report.composition.RuntimeViewsheet;
+import inetsoft.report.composition.WorksheetEngine;
+import inetsoft.report.internal.Util;
+import inetsoft.uql.viewsheet.*;
+import inetsoft.uql.viewsheet.internal.*;
+import inetsoft.util.Tool;
+import inetsoft.web.viewsheet.command.OpenAnnotationFormatDialogCommand;
+import inetsoft.web.viewsheet.event.annotation.OpenAnnotationFormatDialogEvent;
+import inetsoft.web.viewsheet.event.annotation.UpdateAnnotationFormatEvent;
+import inetsoft.web.viewsheet.model.dialog.AnnotationFormatDialogModel;
+import inetsoft.web.viewsheet.service.*;
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.stereotype.Service;
+
+import java.security.Principal;
+
+@Service
+@ClusterProxy
+public class VSAnnotationFormatService {
+
+   public VSAnnotationFormatService(VSObjectService service,
+                                    VSAnnotationService annotationService) {
+      this.service = service;
+      this.annotationService = annotationService;
+   }
+
+   @ClusterProxyMethod(WorksheetEngine.CACHE_NAME)
+   public Void openFormatDialog(@ClusterProxyKey String id, OpenAnnotationFormatDialogEvent event,
```

Comment:
Need to make the event serializable. See https://immutables.github.io/immutable.html#serialization. In general for any arguments that are objects, you need to look at if they can be serialized in order to support being passed as a remote method argument.